sky/packages/sky/lib/src/animation/timeline.dart

```diff
@@ -5,17 +5,89 @@
 import 'dart:async';
 
 import 'package:newton/newton.dart';
-import 'package:sky/src/animation/curves.dart';
 import 'package:sky/src/animation/animated_value.dart';
-import 'package:sky/src/animation/animated_simulation.dart';
+import 'package:sky/src/animation/curves.dart';
+import 'package:sky/src/animation/scheduler.dart';
+
+typedef _TickerCallback(Duration elapsed);
+
+/// Calls its callback once per animation frame
+class _Ticker {
+  /// Constructs a ticker that will call onTick once per frame while running
+  _Ticker(_TickerCallback onTick) : _onTick = onTick;
+
+  final _TickerCallback _onTick;
+
+  Completer _completer;
+  int _animationId;
+  Duration _startTime;
+
+  /// Start calling onTick once per animation frame
+  ///
+  /// The returned future resolves once the ticker stops ticking.
+  Future start() {
+    assert(!isTicking);
+    assert(_startTime == null);
+    _completer = new Completer();
+    _scheduleTick();
+    return _completer.future;
+  }
+
+  /// Stop calling onTick
+  ///
+  /// Causes the future returned by [start] to resolve.
+  void stop() {
+    if (!isTicking)
+      return;
+
+    _startTime = null;
+
+    if (_animationId != null) {
+      scheduler.cancelAnimationFrame(_animationId);
+      _animationId = null;
+    }
+
+    // We take the _completer into a local variable so that !isTicking
+    // when we actually complete the future (isTicking uses _completer
+    // to determine its state).
+    Completer localCompleter = _completer;
+    _completer = null;
+    assert(!isTicking);
+    localCompleter.complete();
+  }
+
+  /// Whether this ticker has scheduled a call to onTick
+  bool get isTicking => _completer != null;
 
-/// A simulation that linearly varies from [begin] to [end] over [duration]
+  void _tick(Duration timeStamp) {
+    assert(isTicking);
+    assert(_animationId != null);
+    _animationId = null;
+
+    if (_startTime == null)
+      _startTime = timeStamp;
+
+    _onTick(timeStamp - _startTime);
+
+    // The onTick callback may have scheduled another tick already.
+    if (isTicking && _animationId == null)
+      _scheduleTick();
+  }
+
+  void _scheduleTick() {
+    assert(isTicking);
+    assert(_animationId == null);
+    _animationId = scheduler.requestAnimationFrame(_tick);
+  }
+}
+
+/// A simulation that varies from [begin] to [end] over [duration] using [curve]
```

Comment:
should we rename "simulation"? It's not clear to me that something as simple as "apply a function to a variable" should be called a "simulation".

How does this class differ from an AnimatedValue<double>? As far as I can tell, all the logic in this class can be summed up in two lines:
```
_tween = new AnimatedValue<double>(begin, end: end, curve: curve);
// ...
_tween.setProgress(t, Direction.forward);
```